# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/gtk/nsWaylandDisplay.cpp
# Commit: bf6209fd39e8
# Full Hash: bf6209fd39e85664974d19bfc1c8cb6d06d48a21
# Author: stransky <stransky@redhat.com>
# Date: 2025-02-05 16:56:59
# Regressor Bug: 1944490
# File Overlap Count: 1
# Description:
#   Bug 1944490 [Linux] Ensure DMABufFormats provides basic dmabuf formats r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D236688
# ==============================================================================

diff -r a374b7104d05 -r bf6209fd39e8 widget/gtk/nsWaylandDisplay.cpp
--- a/widget/gtk/nsWaylandDisplay.cpp	Wed Feb 05 11:23:30 2025 +0000
+++ b/widget/gtk/nsWaylandDisplay.cpp	Wed Feb 05 11:26:53 2025 +0000
@@ -413,7 +413,6 @@
     return;
   }
   mDmabuf = aDmabuf;
-  mFormats = new DMABufFormats();
   mDmabufIsFeedback =
       (aVersion >= ZWP_LINUX_DMABUF_V1_GET_DEFAULT_FEEDBACK_SINCE_VERSION);
   if (mDmabufIsFeedback) {
@@ -423,6 +422,13 @@
   }
 }
 
+void nsWaylandDisplay::EnsureDMABufFormats() {
+  if (mDmabuf && !mDmabufIsFeedback) {
+    mFormats->InitV3Done();
+  }
+  mFormats->EnsureBasicFormats();
+}
+
 void nsWaylandDisplay::SetXdgActivation(xdg_activation_v1* aXdgActivation) {
   mXdgActivation = aXdgActivation;
 }
@@ -646,13 +652,12 @@
   // in a similar fashion
   wl_log_set_handler_client(WlLogHandler);
 
+  mFormats = new DMABufFormats();
   mRegistry = wl_display_get_registry(mDisplay);
   wl_registry_add_listener(mRegistry, &registry_listener, this);
   wl_display_roundtrip(mDisplay);
   wl_display_roundtrip(mDisplay);
-  if (mDmabuf && !mDmabufIsFeedback) {
-    mFormats->InitV3Done();
-  }
+  EnsureDMABufFormats();
 
   for (auto& e : mSupportedTransfer) {
     e = -1;